### PR TITLE
feat(enhancement): Make inscrutable on player ships useful

### PR DESCRIPTION
**Pull request by TomGoodIdea**
**Feature:** This PR implements the feature request detailed and discussed in issue #5544
## Feature Details
When a player ship with `inscrutable` attribute gets scanned by an NPC, a special scan message is displayed. Even if the ship has some illegal outfits or cargo, fines aren't applied and stealth missions don't fail.

## Additional notes
While testing this, I saw some scan messages overlapping another messages, leaving some dim letters of the original messages. Not sure what causes such behavior or if it was happening before.

### CODE REVIEW

**Comment by Zitchas** [comment]
It shouldn't be too often the player is flying an inscrutable ship.... But it's good to make everything work properly, and close out an old issue in the process. Thanks!

**Review by quyykk — APPROVED** [review]
LGTM

I'm not totally sure the message is clear, I find "completed ... with no useful results" to be a bit weird of a phrasing. I think I would use "failed its scan of your cargo." instead, but I'll let someone else weigh in

**Comment by Zitchas** [comment]
I'm not sure inscrutable should be applied to cargo, honestly. Outfits and the ship itself, sure. But cargo... I don't know. A ship being inscrutable is supposed to be the ship is incomprehensible; not that it is utterly inpenetrable. Although I suppose it could be.

**Comment by quyykk** [comment]
You are right, `inscrutable` only applies to outfit scans, not cargo scans! For some reason I thought it also applied to cargo, but no. So yeah don't make it apply to cargo scans.

File: source/Politics.cpp [hunk 1]
```diff
@@ -50,7 +50,8 @@ namespace {
 	// Check if the ship evades being outfit scanned.
 	bool EvadesOutfitScan(const Ship &ship)
 	{
-		return Random::Real() > 1. / (1. + ship.Attributes().Get("scan interference"));
+		return ship.Attributes().Get("inscrutable") ||
```
Comment (quyykk) [inline source/Politics.cpp:53]:
```suggestion
		return ship.Attributes().Get("inscrutable") > 0. ||
```

let's check whether this is in fact positive.

**Review by quyykk — APPROVED** [review]

**Review by warp-core — APPROVED** [review]

**Pull request merged**